source/framework/core/inc/TRestDataSet.h

```diff
@@ -85,6 +88,8 @@ class TRestDataSet : public TRestMetadata {
     Int_t GetNumberOfColumns() { return fDataSet.GetColumnNames().size(); }
     Int_t GetNumberOfBranches() { return GetNumberOfColumns(); }
 
+    Double_t GetTotalTimeInSeconds() { return fTotalDuration; }
```

Comment:
Yes, I replied there, I think that it would be easy once the dataset has loaded to integrate those periods time, subtract the time from total time, and remove those events using a standard TCut or Filter to replace the existing DataFrame/TTree. 

I added this issue to #346 in case you wish to take over